### PR TITLE
Handle different files for iOS/tvOS

### PR DESCRIPTION
**Pull request by igoriols**
# Goal
Be able to build iOS and tvOS framerworks on the same project.

# How to test
- Run `Clappr_Tests`, `Clappr_UITests` and `Clappr-tvOS-Tests`. 
- Copy the generated framework for tvOS and embed in a sample project (can be a similar sample app used in tvOS branch) and run the app.

### CODE REVIEW

**Comment by igoriols** [comment]
[[iOS&tvOS | Clappr] Tratar arquivos modificados no tvOS](https://trello.com/c/7uT5JwO1/16-iostvos-clappr-tratar-arquivos-modificados-no-tvos)

File: Clappr-tvOS/Classes/Base/Options.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
+public typealias Options = [String: Any]
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/Options.swift]: You can extract the options of tvOS to shared options, what you think?

```swift
#if os(tvOS)
public let kAutoPlay = "autoplay"
public let kLoop = "loop"
public let kMetaData = "metadata"
public let kMetaDataTitle = "mdTitle"
public let kMetaDataDescription = "mdDescription"
public let kMetaDataDate = "mdDate"
public let kMetaDataArtwork = "mdArtwork"
#endif
```

File: Clappr-tvOS/Classes/Base/MediaOption.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+open class MediaOption: Equatable {
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/MediaOption.swift]: This file is exactly the same on shared target

File: Clappr-tvOS/Classes/Base/Playback.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    fileprivate(set) open var subtitles: [MediaOption]?
+    fileprivate(set) open var audioSources: [MediaOption]?
+
+    internal(set) open var options: Options
+
+    open var source: String? {
+        return options[kSourceUrl] as? String
+    }
+
+    open var autoPlay: Bool {
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/Playback.swift]: This is one of three differences between the shared playback, what you think about adding a pre-processor to reuse code?

File: Clappr-tvOS/Classes/Base/Playback.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    open override func render() {
+        once(Event.ready.rawValue) { [unowned self] _ in
+            if self.startAt != 0.0 {
+                self.seek(self.startAt)
+            }
+
+            self.selectDefaultMediaOptions()
+        }
+
+        if autoPlay {
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/Playback.swift]: This is one of three differences between the shared playback, what you think about adding a pre-processor to reuse code

File: Clappr-tvOS/Classes/Base/Playback.swift [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            let defaultAudioSource = audioSources?.filter({ $0.language == defaultAudioLanguage }).first {
+            selectedAudioSource = defaultAudioSource
+        }
+    }
+
+    open func play() {}
+    open func pause() {}
+    open func stop() {}
+    open func seek(_: TimeInterval) {}
+    open func mute(_: Bool) {}
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/Playback.swift]: This is one of three differences between the shared playback, what you think about adding a pre-processor to reuse code

File: Clappr-tvOS/Classes/Base/Container.swift [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+open class Container: UIBaseObject {
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Base/Container.swift]: This class doesn't have too much difference between the shared one

File: Clappr-tvOS/Classes/Plugin/Playback/NoOpPlayback.swift [hunk 7]
```diff
@@ -0,0 +1,53 @@
+open class NoOpPlayback: Playback {
+    fileprivate var errorLabel = UILabel(frame: CGRect.zero)
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Plugin/Playback/NoOpPlayback.swift]: This class doesn't have too much difference between the shared one, what do you think about merging it?

File: Clappr-tvOS/Classes/Formatter/ClapprDateFormatter.swift [hunk 8]
```diff
@@ -0,0 +1,11 @@
+open class ClapprDateFormatter {
+    fileprivate static let hourInSeconds: Double = 1 * 60 * 60
```
Comment (CristianDRM) [inline Clappr-tvOS/Classes/Formatter/ClapprDateFormatter.swift]: This class could be easily merged into the shared one.

File: Clappr-tvOS/Classes/Base/FullscreenController.swift [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import UIKit
+
+class FullscreenController: UIViewController {
+
+    #if !os(tvOS)
```
[the remark this answers is not in the document]
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/FullscreenController.swift]: Hey @CristianDRM and @jchahoud , `FullScreenController`is used in `MediaControl`. Actually we don't need `MediaControl` for tvOS yet. We can change to optional the `mediaControl` property present in `Loader` and initialize it only for iOS. If we do that, we need to update where it is used (few places). Doing that, we can have only one `MediaControl`and `FullScreenController` classes, only used by iOS target.
What you think about it?

File: Clappr-tvOS/Classes/Base/MediaControl.swift [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+            image = livePlayback ? stopButtonImage : pauseButtonImage
+        } else {
+            image = playButtonImage
+        }
+
+        playbackControlButton?.setImage(image, for: UIControlState())
+    }
+
+    open func bindOrientationChangedListener() {
+        #if !os(tvOS)
```
[the remark this answers is not in the document]
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/MediaControl.swift]: If we discard the `MediaControl` for tvOS, we can discard it.

File: Clappr-tvOS/Classes/Base/MediaOption.swift (hunk 2 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/MediaOption.swift]: @CristianDRM thanks, I'm going to remove the duplicity.

File: Clappr-tvOS/Classes/Base/Loader.swift [hunk 11]
```diff
@@ -0,0 +1,63 @@
+open class Loader {
+    internal(set) open var playbackPlugins: [Plugin.Type] = [AVFoundationPlayback.self, NoOpPlayback.self]
+    #if os(iOS)
```
[the remark this answers is not in the document]
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/Loader.swift]: Done!

File: Clappr-tvOS/Classes/Base/MediaControl.swift (hunk 10 above)
[the remark this answers is not in the document]
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/MediaControl.swift]: For now, I'm going to remove this method. In the future, we can apply the change I mentioned above.

File: Clappr-tvOS/Classes/Base/Options.swift (hunk 1 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/Options.swift]: Hey @CristianDRM I'm going to change this in the future along with the other changes, ok?

File: Tests/Clappr-tvOS-Tests/ContainerTests.swift [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+                it("Should add container plugins from loader") {
+                    loader.addExternalPlugins([FakeContainerPlugin.self, AnotherFakeContainerPlugin.self])
+
+                    let container = Container(loader: loader, options: options)
+
+                    expect(container.hasPlugin(FakeContainerPlugin.self)).to(beTrue())
+                    expect(container.hasPlugin(AnotherFakeContainerPlugin.self)).to(beTrue())
+                }
+
+                #if os(iOS)
```
[the remark this answers is not in the document]
Comment (igoriols) [reply Tests/Clappr-tvOS-Tests/ContainerTests.swift]: Thanks, I removed the test.

File: Clappr-tvOS/Classes/Base/Playback.swift (hunk 3 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/Playback.swift]: Hey @CristianDRM, I created a card to do this change in another moment.

File: Clappr-tvOS/Classes/Base/Playback.swift (hunk 4 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/Playback.swift]: Hey @CristianDRM, I created a card to do this change in another moment.

File: Clappr-tvOS/Classes/Base/Playback.swift (hunk 5 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Base/Playback.swift]: Hey @CristianDRM, I created a card to do this change in another moment.

File: Clappr-tvOS/Classes/Plugin/Playback/NoOpPlayback.swift (hunk 7 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Plugin/Playback/NoOpPlayback.swift]: Hey @CristianDRM, I created a card to do this change in another moment.

File: Clappr-tvOS/Classes/Formatter/ClapprDateFormatter.swift (hunk 8 above)
Comment (igoriols) [reply Clappr-tvOS/Classes/Formatter/ClapprDateFormatter.swift]: Done!

**Pull request merged**